### PR TITLE
(Feature) 88 adminwallet lock based on mutli address

### PR DESCRIPTION
**Pull request by AlexeyKosinski**
# Description
Issue #88 
- change admin wallet to multi admin wallet
- change Mongo queue to work with many addresses

# How Has This Been Tested?
set NUMBER_OF_ADMIN_WALLET_ACCOUNTS=1 in server .env file
please run script in dapp repo
./loadtest/run-art.sh
choose admin-wallet test
connect to mongo and check what happens with wallets in realtime

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [x] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by sirpy** [comment]
@AlexeyKosinski whats the status here?

File: src/server/utils/tx-manager/queueMongo.js [hunk 1]
```diff
@@ -244,11 +219,10 @@ export default class queueMongo {
   async run() {
     try {
       if (this.queue.length > 0) {
-        const nextTr = this.queue[0]
+        const nextTr = this.queue.shift()
 
         const walletNonce = await this.getWalletNonce(nextTr.addresses)
         if (walletNonce) {
-          this.queue.shift()
```
Comment (sirpy) [inline src/server/utils/tx-manager/queueMongo.js]: What if there's an exception in the above await? you need to restore the TX to the queue

Comment (AlexeyKosinski) [reply src/server/utils/tx-manager/queueMongo.js]: it was the main problem why it is not working with many wallets... ok  I will add try catch which return it to queue

**Comment by AlexeyKosinski** [comment]
I've tried to use fuse network with `NUMBER_OF_ADMIN_WALLET_ACCOUNTS=1` but it's generates admin wallet 0xc736793ff31e04807cbf20b39d50ac7a04a4bdad with balance 0.
also, I've tried another wallet with non zero balance but all transactions was reverted by contract (I think because it is not admin address) so to make it works on fuse we need add new admin wallet address to smartcontract

File: src/server/utils/tx-manager/queueMutex.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ export default class queueMutex {
     this.nonce = null
     this.mutex = new Mutex()
     this.lastFail = null
+    this.getTransactionCount = () => 0
```
Comment (yaroslav-fedyshyn-nordwhale) [inline src/server/utils/tx-manager/queueMutex.js]: @AlexeyKosinski What for do you create this function here? Is it really needed?

Comment (AlexeyKosinski) [reply src/server/utils/tx-manager/queueMutex.js]: it is just for initial value. when we init adminWallet we update this function to real web3 function

File: src/server/utils/tx-manager/queueMongo.js [hunk 3]
```diff
@@ -178,14 +178,14 @@ export default class queueMongo {
     try {
       if (this.queue.length > 0) {
         nextTr = this.queue.shift()
-
         const walletNonce = await this.getWalletNonce(nextTr.addresses)
         if (walletNonce) {
           nextTr.cb({ nonce: walletNonce.nonce, address: walletNonce.address })
+        } else {
+          this.queue.splice(0, 0, nextTr)
```
Comment (sirpy) [inline src/server/utils/tx-manager/queueMongo.js]: why not push?unshift?

Comment (sirpy) [reply src/server/utils/tx-manager/queueMongo.js]: why not also exception?

Comment (AlexeyKosinski) [reply src/server/utils/tx-manager/queueMongo.js]: I want to insert it into the begin of the queue. yes we can use unshift as well as a splice.
If we add unshift to the catch can get a problem with the infinity loop. example:
1 get from queue
2 run
3 catch and return to the queue
4 go to `1`

File: src/server/blockchain/AdminWallet.js [hunk 4]
```diff
@@ -78,91 +78,81 @@ export class Wallet {
...
+          gas: 500000,
+          gasPrice: web3Utils.toWei('1', 'gwei')
+        }
+      )
+      this.reserveContract = new this.wallet.web3.eth.Contract(
+        ReserveABI.abi,
+        get(ContractsAddress, `${this.network}.GoodDollarReserve`, ReserveABI.networks[this.networkId].address),
+        {
+          from: this.address,
+          gas: 500000,
```
Comment (StanislavShevchenko) [inline src/server/blockchain/AdminWallet.js]: maybe constant

File: src/server/blockchain/AdminWallet.js [hunk 5]
```diff
@@ -78,91 +80,81 @@ export class Wallet {
...
-      this.web3.eth.accounts.wallet.add(account)
-      this.web3.eth.defaultAccount = account.address
-      this.address = account.address
-      log.debug('Initialized by private key:', account.address)
-    } else if (conf.mnemonic) {
-      let root = HDKey.fromMasterSeed(bip39.mnemonicToSeed(this.mnemonic))
-      var path = "m/44'/60'/0'/0/0"
-      let addrNode = root.derive(path)
-      let account = this.web3.eth.accounts.privateKeyToAccount('0x' + addrNode._privateKey.toString('hex'))
-      this.web3.eth.accounts.wallet.add(account)
```
Comment (sirpy) [inline src/server/blockchain/AdminWallet.js]: here just loop over the paths and add the accounts
no need to copy the code from dapp

**Pull request merged**